01_Data/src/layers/sequelize/repository/Tariff.ts

```diff
@@ -22,30 +30,36 @@ export class SequelizeTariffRepository extends SequelizeRepository<Tariff> imple
     });
   }
 
-  async createOrUpdateTariff(tariff: Tariff): Promise<Tariff> {
+  async upsertTariff(tariff: Tariff): Promise<Tariff> {
     return await this.s.transaction(async (transaction) => {
-      const savedTariff = await this.s.models[Tariff.MODEL_NAME].findOne({
-        where: {
-          stationId: tariff.stationId,
-          unit: tariff.unit,
-        },
-        transaction,
+      const savedTariff = await this.readOnlyOneByQuery({
+        where: {id: tariff.id, countryCode: tariff.countryCode, partyId: tariff.partyId},
+        include: TariffElement,
+        transaction
       });
-      if (savedTariff) {
-        return (await this.updateByKey({ ...tariff }, savedTariff.dataValues.id)) as Tariff;
+
+      if (!savedTariff) {
+        const createdTariff = await tariff.save({ transaction });
+        this.emit('created', [createdTariff]);
+        return createdTariff;
       }
-      const createdTariff = await tariff.save({ transaction });
-      this.emit('created', [createdTariff]);
-      return createdTariff;
+
+      // TODO: optimize
```

Comment:
Keeping it simple for now; we can consider optimizations later, if that works for everyone?